### PR TITLE
Allow handshake errors to be reported with CONNECTION_CLOSE

### LINKED ISSUE #608: Handshake errors
If the handshake fails, how do we signal that?  We could send CONNECTION_CLOSE in the clear...

If version negotiation fails, how do we signal that?  I don't think that we can, though we currently require that.

### PR DESCRIPTION
**Pull request by martinthomson**
Closes #608.

### CODE REVIEW

**Comment by RyanAtGoogle** [comment]
Nice. LGTM

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -1287,6 +1287,13 @@ Similarly, there is no need to increase the number of allowed streams until the
 handshake completes.
 
 
+### Handshake Failures
+
+The `CONNECTION_CLOSE` frame MAY be sent by either endpoint during the handshake
+to signal a fatal error with connection establishment.  A `STREAM` frame
+carrying a TLS alert MAY be included in the same packet.
+
```
Comment (huitema) [inline draft-ietf-quic-tls.md]: Maybe we should add a note in the next section, "Denial of Service with Unprotected Packets". Sending "connection close" frames in Server Clear Text or Client Clear Text packets would be another way to deny connection.

Comment (martinthomson) [reply draft-ietf-quic-tls.md]: I'll do that separately: see https://github.com/quicwg/base-drafts/pull/908/files

**Comment by martinthomson** [comment]
@seanturner, PTAL

**Comment by MikeBishop** [comment]
Why is this in the TLS draft?  If we're allowing CONNECTION_CLOSE in Handshake packets, shouldn't that be in the Transport doc?

**Comment by martinthomson** [comment]
The TLS doc contains all the packet protection stuff, which includes rules about what can be sent and when.  The plan is to move this text to -transport eventually.

**Pull request merged**